Testlio/testlio-automation-templates
# Move lib to jar

**Related PRs**
- https://github.com/Testlio/java-automation-framework/pull/1

**Testing strategy**

If needed, refer to the User Guide in order to resolve any technical issues with running tests locally (for example, if you never configured your env to run tests locally): https://testlions.atlassian.net/wiki/spaces/GREEN/pages/3149234177/Test+Automation+User+Guide

**Test A. Running tests locally**
1. Navigate to the root directory of this repository folder in Terminal locally
2. Install Testlio framework library to your local Maven repository using command:
```
mvn install:install-file \
   -Dfile=java-automation-framework-1.0-jar-with-dependencies.jar \
   -DgroupId=com.testlio \
   -DartifactId=java-automation-framework \
   -Dversion=1.0 \
   -Dpackaging=jar \
   -DgeneratePom=true
```
3. Navigate to the `testlio-web-java-testng-template` subfolder
4. Add Testlio user's credentials to the `src/test/resources/credentials.properties`
5. Run browser tests locally using command:
```
mvn clean test \
  -Dprovider=local \
  -Dexecution.type=web \
  -Dbrowser.name=chrome
```
6. Check that tests executed successfully
7. **Optional:** you can try to perform the same steps for `testlio-mobile-java-testng-template`, but you will have to configure Android/iOS simulator or a real device to run them locally.

**Test B. Tests packaging**
1. Install library as described in steps 1-2 of Test A
2. Navigate to the `testlio-web-java-testng-template` subfolder
3. Package script using following command:
```
mvn package -DskipTests
```
4. If command executed successfully, you should be able to find packaged script in the directory `<project root>/target/testlio-web-java-testng-template-*****.zip`
5. Try to use this package to schedule the desktop/mobile web run on platform
6. **Optional:** you can try to perform the same steps for `testlio-mobile-java-testng-template`. Steps are the same.
